gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorTimeBasedJobPropCreator.java

```diff
@@ -131,11 +145,36 @@ private boolean shouldProcessFolder(Path inputFolder, Path outputFolder, DateTim
       LOG.info(String.format("Folder time for %s is %s, later than the latest allowed folder time, %s. Skipping",
           inputFolder, folderTime, latestAllowedFolderTime));
       return false;
-    } else if (folderAlreadyCompacted(outputFolder)) {
-      LOG.info(String.format("Folder %s already compacted. Skipping", outputFolder));
-      return false;
+    } else {
+      return true;
+    }
+  }
+
+  /**
+   * Return job properties for a job to handle the appearance of data within inputFolder which is
+   * more recent than the time of the last compaction.
+   */
+  private State createJobPropsForLateData(Path inputFolder, Path outputFolder, Path tempFolder,
+      List<Path> newDataFiles) throws IOException {
+    if (this.state.getPropAsBoolean(ConfigurationKeys.COMPACTION_RECOMPACT_FOR_LATE_DATA,
+        ConfigurationKeys.DEFAULT_COMPACTION_RECOMPACT_FOR_LATE_DATA)) {
+      LOG.info(String.format("Deleting output folder at %s and recompacting.", outputFolder));
+      this.fs.delete(outputFolder, true);
```

Comment:
Don't delete outputFolder here. It is deleted in `moveTmpPathToOutputPath()`. If it is deleted here, the output folder will be unavailable during the re-compaction which is not desirable.